airbyte-integrations/connectors/source-github/source_github/streams.py

```diff
@@ -0,0 +1,452 @@
+
+            # This whole try/except situation in `read_records()` isn't good but right now in `self._send_request()`
+            # function we have `response.raise_for_status()` so we don't have much choice on how to handle errors.
+            # We added this try/except code because for private repositories `Teams` stream is not available and we get
+            # "404 Client Error: Not Found for url: https://api.github.com/orgs/sherifnada/teams?per_page=100" error.
+            if "/teams?" in error_msg:
+                error_msg = f"Syncing Team stream isn't available for repository {self.repository}"
+
+            self.logger.warn(error_msg)
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+
+        params = {"per_page": self.page_size}
+
+        if next_page_token:
+            params.update(next_page_token)
+
+        return params
+
+    def request_headers(self, **kwargs) -> Mapping[str, Any]:
+        # Without sending `User-Agent` header we will be getting `403 Client Error: Forbidden for url` error.
+        return {
+            "User-Agent": "PostmanRuntime/7.28.0",
+        }
+
+    def parse_response(
+        self,
+        response: requests.Response,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping]:
+        for record in response.json():  # GitHub puts records in an array.
+            self.transform(record=record)
+            yield record
+
+    def transform(self, record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Use this method to:
+            - remove excessive fields from record;
+            - minify subelements in the record. For example, if you have `reviews` record which looks like this:
+            {
+              "id": 671782869,
+              "node_id": "MDE3OlB1bGxSZXF1ZXN0UmV2aWV3NjcxNzgyODY5",
+              "user": {
+                "login": "keu",
+                "id": 1619536,
+                ... <other fields>
+              },
+              "body": "lgtm, just  small comment",
+              ... <other fields>
+            }
+
+            `user` subelement contains almost all possible fields fo user and it's not optimal to store such data in
+            `reviews` record. We may leave only `user.id` field and save in to `user_id` field in the record. So if you
+            need to do something similar with your record you may use this method.
+        """
+        for field in self.object_fields:
+            field_value = record.pop(field, None)
+            record[f"{field}_id"] = field_value.get("id") if field_value else None
+
+        for field in self.list_fields:
+            field_values = record.pop(field, [])
+            record[field] = [value["id"] for value in field_values]
```

Comment:
```suggestion
            record[field] = [value.get("id") for value in field_values]
```
Let`s be a little more safely))

Reply:
Done.